pkg/config/config.go

```diff
@@ -408,51 +341,46 @@ func (dc *DeploymentConfig) ExpandConfig() error {
 	return nil
 }
 
-func (dc *DeploymentConfig) addModuleConnection(ref reference, kind connectionKind, sharedVariables []string) error {
-	if dc.moduleConnections == nil {
-		dc.moduleConnections = make(map[string][]ModConnection)
-	}
-
-	if !kind.IsValid() {
-		log.Fatal(unexpectedConnectionKind)
+func (b *Blueprint) setGlobalLabels() {
+	if !b.Vars.Has("labels") {
+		b.Vars.Set("labels", cty.EmptyObjectVal)
 	}
-
-	conn := ModConnection{
-		ref:             ref,
-		kind:            kind,
-		sharedVariables: sharedVariables,
-	}
-
-	fromModID := ref.FromModuleID()
-	dc.moduleConnections[fromModID] = append(dc.moduleConnections[fromModID], conn)
-	return nil
 }
 
-// GetModuleConnections returns the graph of connections between modules and
-// from modules to deployment variables
-func (dc *DeploymentConfig) GetModuleConnections() map[string][]ModConnection {
-	return dc.moduleConnections
-}
+// listUnusedModules provides a list modules that are in the
+// "use" field, but not actually used.
+func (m Module) listUnusedModules() []string {
+	used := map[string]bool{}
+	cty.Walk(m.Settings.AsObject(), func(p cty.Path, v cty.Value) (bool, error) {
```

Comment:
I think it is worth leaving a comment here and at other invocations of cty.Walk guiding future readers as to what is happening. 1-liner is good. Something like"

```go
// Recurse through objects/maps/lists evaluating each element for references marked by Use keyword
```

Reply:
I think you are familiar enough with the idea that cty.Value abstracts many types of values, but future readers will not be.